flink-table/flink-table-planner/src/test/java/org/apache/flink/table/catalog/CatalogFunctionTestBase.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.factories.utils.TestCollectionTableFactory;
+import org.apache.flink.table.functions.ScalarFunction;
+import org.apache.flink.types.Row;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link CatalogFunction}.
+ */
+public abstract class CatalogFunctionTestBase {
+	private static TableEnvironment tableEnv;
+
+	public static void setTableEnv(TableEnvironment e) {
+		tableEnv = e;
+	}
+
+	public abstract void execute() throws Exception;
+
+	@Test
+	public void testCreateCatalogFunctionInDefaultCatalog() {
+		String ddl1 = "create function f1 as 'org.apache.flink.function.TestFunction'";
+		tableEnv.sqlUpdate(ddl1);
+		assertTrue(Arrays.asList(tableEnv.listFunctions()).contains("f1"));
+
+		tableEnv.sqlUpdate("DROP FUNCTION IF EXISTS default_catalog.default_database.f1");
+		assertTrue(!Arrays.asList(tableEnv.listFunctions()).contains("f1"));
+	}
+
+	@Test
+	public void testCreateFunctionWithFullPath() {
+		String ddl1 = "create function default_catalog.default_database.f2 as" +
+			" 'org.apache.flink.function.TestFunction'";
+		tableEnv.sqlUpdate(ddl1);
+		assertTrue(Arrays.asList(tableEnv.listFunctions()).contains("f2"));
+
+		tableEnv.sqlUpdate("DROP FUNCTION IF EXISTS default_catalog.default_database.f2");
+		assertTrue(!Arrays.asList(tableEnv.listFunctions()).contains("f2"));
+	}
```

Comment:
Use `assertFalse` instead. BTW, the test name is "testCreate" but you also test "Drop Function".

Reply:
Done.